tests/unit/mathematics/linalg/Linalg_dot_unittest.cc

```diff
@@ -0,0 +1,60 @@
+#include <shogun/lib/config.h>
+
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+
+#ifdef HAVE_VIENNACL
+#include <shogun/mathematics/linalg/LinalgBackendViennaCL.h>
+#endif
+
+#include <shogun/lib/SGVector.h>
+#include <gtest/gtest.h>
+
+using namespace shogun;
+TEST(LinalgDot, linalg_cpu_backend_dot)
```

Comment:
Never mind..we also have `gpu_dot_without_gpu_backend` in this file.
What is the correct name for that? Also go with `LinalgBackendEigen` or just use `Linalg` for all?